tests/cypress/support/commands.ts

```diff
@@ -398,3 +414,36 @@ Cypress.Commands.add('deleteRole', (roleName, roleTypeTemplate) => {
   // Verify that there are no rows
   cy.contains('There are no rows which match your search query.', { timeout: 2000 }).should('be.visible');
 })
+
+// Add label to the imported cluster(s)
+Cypress.Commands.add('assignClusterLabel', (clusterName, key, value) => {
+  cy.filterInSearchBox(clusterName);
+  cy.open3dotsMenu(clusterName, 'Edit Config');
+  cy.clickButton('Add Label');
+  cy.get('div[class="row"] div[class="key-value"] input[placeholder="e.g. foo"]').last().type(key);
+  cy.get('div[class="row"] div[class="key-value"] textarea[placeholder="e.g. bar"]').last().type(value);
+  cy.wait(500);
+  cy.clickButton('Save');
+  cy.contains('Save').should('not.exist');
+})
+
+// Create clusterGroup based on label assigned to the cluster
+Cypress.Commands.add('createClusterGroup', (clusterGroupName, key, value, lastClusterName, clusterCount=1) => {
+  cy.fleetNamespaceToggle('fleet-default');
+  cy.clickButton('Create');
+  cy.get('input[placeholder="A unique name"]').type(clusterGroupName);
+  cy.clickButton('Add Rule');
+  cy.get('[data-testid="input-match-expression-key-control-0"]').focus().type(key);
+  cy.get('[data-testid="input-match-expression-values-control-0"]').type(value);
+  // Pass the clusterCount, when there are more than 1 cluster to match.
+  cy.get('div.banner.success').contains(`Matches ${clusterCount} of 3 existing clusters, including "${lastClusterName}"`).should('be.visible');
```

Comment:
Please do not hardcode the expected amount of clusters here `... 3 existing clusters`

Reply:
If that is the case, I need to get the banner and assert it using `expect` block.

Reply:
Also not sure what will happen when all clusters are added into that group, I see different string `Matches all X existing clusters`

Reply:
@thehejik Should I pass assert string from the test only and match here ?